pulsar-client-api/src/main/java/org/apache/pulsar/client/api/Range.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.api;
+
+/**
+ * Int range.
+ */
+public class Range {
+
+    private final int start;
+    private final int end;
+
+
+    public Range(int start, int end) {
+        if (end < start) {
+            throw new IllegalArgumentException("Range end must >= range start.");
+        }
+        this.start = start;
+        this.end = end;
+    }
+
+    public static Range of(int start, int end) {
+        return new Range(start, end);
+    }
+
+    public int getStart() {
+        return start;
+    }
+
+    public int getEnd() {
+        return end;
+    }
+
+    public Range intersect(Range range) {
```

Comment:
Guava has a `Range` class. You might consider reusing its class.

Reply:
The pulsar client api module do not add guava dependency, shall we add the guava dependency?